app/jobs/api_fetches_harry_potter_movies_job.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ApiFetchesHarryPotterMoviesJob < ApplicationJob
+  queue_as :api_save_harry_potter
+
+  def perform
+    reschedule_job
+    do_work
+  end
+
+  def do_work
+    api_search = Contexts::Services::Imdb::Search.new
+    data = api_search.call(expression: 'Harry Potter')
+    json = Contexts::Services::Imdb::Parse.new(response: data).call
+    Contexts::Services::Imdb::Save.new(response: json).call
+  end
```

Comment:
generally job should looks like this: 
```suggestion
  def perform
    api_search = Contexts::Services::Imdb::Search.new
    data = api_search.call(expression: 'Harry Potter')
    json = Contexts::Services::Imdb::Parse.new(response: data).call
    Contexts::Services::Imdb::Save.new(response: json).call
  end
```
then when calling a job, you may specify parameters like 
`ApiFetchesHarryPotterMoviesJob.set(wait: 1.hour).perform_later`